pandas/core/arrays/integer.py

```diff
@@ -391,7 +392,7 @@ def astype(self, dtype, copy=True):
 
         # coerce
         data = self._coerce_to_ndarray()
-        return data.astype(dtype=dtype, copy=False)
+        return astype_nansafe(data, dtype, copy=None)
```

Comment:
This was necessary so that 

```python
In [8]: s.values
Out[8]: IntegerArray([0, nan, 2], dtype='Int8')

In [9]: s.astype('uint32')
```

still raises. Previously, it did

```python
In [4]: pd.core.arrays.IntegerArray([1, None, 2], dtype='uint8').astype('uint32')
Out[4]: array([         1, 2415919104,          2], dtype=uint32)
```

which I don't think we want. This was only tested at the `Series[IntegerArray].astype` level, which never called EA.astype

Reply:
Can you add a test for this?